packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue

```diff
@@ -331,15 +313,17 @@ onMounted(async () => {
 		isLoadingStructuralComparisons.value = false;
 	}
 
-	props.node.inputs.forEach((input) => {
-		if (input.value) {
-			addModelForComparison(input.value[0]);
-		}
-	});
-
-	const modelIds = props.node.inputs
+	const modelIds: string[] = props.node.inputs
 		.filter((input) => input.status === WorkflowPortStatus.CONNECTED)
 		.map((input) => input.value?.[0]);
+
+	modelsToCompare.value = (await Promise.all(modelIds.map(async (modelId) => getModel(modelId)))).filter(
+		Boolean
```

Comment:
What does the `filter(Boolean)` do?

Reply:
removes null values, ensures that its of type `Model[]` instead of `(Model | null)[]`